dipy/segment/tissue.py

```diff
@@ -5,57 +5,58 @@
 
 
 class TissueClassifierHMRF:
-    r"""
-    This class contains the methods for tissue classification using the Markov
-    Random Fields modeling approach
+    """
+    This class contains the methods for tissue classification using the
+    Markov Random Fields modeling approach.
     """
 
     def __init__(self, save_history=False, verbose=True):
-
         self.save_history = save_history
         self.segmentations = []
         self.pves = []
         self.energies = []
         self.energies_sum = []
         self.verbose = verbose
 
-    def classify(self, image, nclasses, beta, tolerance=None, max_iter=None):
-        r"""
+    def classify(self, image, nclasses, beta, tolerance=1e-05, max_iter=100):
+        """
         This method uses the Maximum a posteriori - Markov Random Field
-        approach for segmentation by using the Iterative Conditional Modes and
-        Expectation Maximization to estimate the parameters.
+        approach for segmentation by using the Iterative Conditional Modes
+        and Expectation Maximization to estimate the parameters.
 
         Parameters
         ----------
         image : ndarray,
-                3D structural image.
+            3D structural image.
         nclasses : int,
-                number of desired classes.
+            Number of desired classes.
         beta : float,
-                smoothing parameter, the higher this number the smoother the
-                output will be.
-        tolerance: float,
-                value that defines the percentage of change tolerated to
-                prevent the ICM loop to stop. Default is 1e-05.
-        max_iter : float,
-                fixed number of desired iterations. Default is 100.
-                If the user only specifies this parameter, the tolerance
-                value will not be considered. If none of these two
-                parameters
+            Smoothing parameter, the higher this number the smoother the
+            output will be.
+        tolerance: float, optional
+            Value that defines the percentage of change tolerated to
+            prevent the ICM loop to stop. Default is 1e-05.
```

Comment:
Remove defaults here: this is difficult to maintain.

Reply:
The default values were following what it already had. If we just remove the default values, it could result in breaking many user code in the next version. Plus, tolerance value is not something a user would change a lot, and is hard to understand how to select the appropriate value.

Reply:
Only meant removing defaults in the documentation, same as in e.g.
https://github.com/dipy/dipy/pull/2707#discussion_r1071309787

If for some reason the default changes in the method signature, changes that the docstring goes out of sync are high.

Reply:
Makes sense, especially since it is not a value user would have to understand most of the time.